pkg/local_object_storage/writecache/delete.go

```diff
@@ -1,51 +1,85 @@
 package writecache
 
 import (
+	"fmt"
+
 	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/common"
 	storagelog "github.com/nspcc-dev/neofs-node/pkg/local_object_storage/internal/log"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
 	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
 	"go.etcd.io/bbolt"
 )
 
 // Delete removes object from write-cache.
 //
 // Returns an error of type apistatus.ObjectNotFound if object is missing in write-cache.
+// Returns ErrReadOnly if the object can't be removed due to read-only mode.
 func (c *cache) Delete(addr oid.Address) error {
-	c.modeMtx.RLock()
```

Comment:
`modeMtx` should be taken for the whole duration -- it also serves as a synchronization point for `Close`, `Flush` etc.

Reply:
Not actualy anymore.